apiserver/remotefirewaller/firewaller.go

```diff
@@ -0,0 +1,63 @@
+// Copyright 2017 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package remotefirewaller
+
+import (
+	"github.com/juju/juju/apiserver/common"
+	"github.com/juju/juju/apiserver/facade"
+	"github.com/juju/juju/apiserver/params"
+	"github.com/juju/juju/feature"
+	"github.com/juju/juju/state/watcher"
+)
+
+func init() {
+	common.RegisterStandardFacadeForFeature("RemoteFirewaller", 1, NewStateRemoteFirewallerAPI, feature.CrossModelRelations)
+}
+
+// RemoteFirewallerAPI provides access to the RemoteFirewaller API facade.
+type RemoteFirewallerAPI struct {
+	st         State
+	pool       StatePool
+	resources  facade.Resources
+	authorizer facade.Authorizer
+}
+
+// NewStateRemoteFirewallerAPI creates a new server-side RemoteFirewallerAPI facade.
+func NewStateRemoteFirewallerAPI(ctx facade.Context) (*RemoteFirewallerAPI, error) {
+	return NewRemoteFirewallerAPI(stateShim{ctx.State()}, statePoolShim{ctx.StatePool()}, ctx.Resources(), ctx.Auth())
+}
+
+// NewRemoteFirewallerAPI creates a new server-side RemoteFirewallerAPI facade.
+func NewRemoteFirewallerAPI(
+	st State,
+	pool StatePool,
```

Comment:
No, i was using it in an earlier iteration when model tags were passed in to the WatchSubnets() API. But I think the method of creating an API facade per model is the right way to go as it's how hosted juju works (albeit using redirection in the controller proxy).

I left it there for future use but perhaps it's YAGNI